dali/dfu/dfurun.cpp

```diff
@@ -601,6 +601,44 @@ class CDFUengine: public CInterface, implements IDFUengine
             throw MakeStringException(DFSERR_CreateAccessDenied,"Create permission denied for physical file(s)");
     }
 
+    void checkForeignFilePermissions(IConstDFUfileSpec *fSpec,IFileDescriptor *fd,IUserDescriptor *user,bool write)
+    {
+        unsigned auditflags = (DALI_LDAP_AUDIT_REPORT|DALI_LDAP_READ_WANTED);
+        if (write)
+            auditflags |= DALI_LDAP_WRITE_WANTED;
+
+        StringBuffer logicalName;
+        fSpec->getLogicalName(logicalName);
+
+        SocketEndpoint daliEP;
+        fSpec->getForeignDali(daliEP);
+
+        CDfsLogicalFileName dlfn;
+        dlfn.set(logicalName);
+        dlfn.setForeign(daliEP,false);
+
+        StringBuffer fu,fp;
+        Owned<IUserDescriptor> foreignuserdesc;
+        if (fSpec->getForeignUser(fu,fp))
+        {
+            foreignuserdesc.setown(createUserDescriptor());
+            foreignuserdesc->set(fu.str(),fp.str());
+        }
+        else
+            foreignuserdesc.set(user);
+
+        SecAccessFlags perm = queryDistributedFileDirectory().getDLFNPermissions(dlfn,foreignuserdesc,auditflags);
+
+        bool checkLegacyPhysicalPerms = getGlobalConfigSP()->getPropBool("expert/@failOverToLegacyPhysicalPerms", !isContainerized());
+        if ((!HASREADPERMISSION(perm) || (write&&!HASWRITEPERMISSION(perm))) && checkLegacyPhysicalPerms)
+            perm = queryDistributedFileDirectory().getFDescPermissions(fd,user,auditflags);
+
+        if (!HASREADPERMISSION(perm))
```

Comment:
does a user necessarily need read permissions to write a file?
As it's coded both are needed if write=true, and the error you get (if no read perms) (on a write=true) is "Lookup permission denied .." - that may be confusing.

Reply:
This is copied from the checkPhysicalFilePermissions(). Looks like it assumes that read permission is needed to write a file. I will check !write here.